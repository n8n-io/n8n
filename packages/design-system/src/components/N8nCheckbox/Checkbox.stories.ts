/* tslint:disable:variable-name */
import N8nCheckbox from "./Checkbox.vue";
import { StoryFn } from '@storybook/vue';
import { action } from '@storybook/addon-actions';

export default {
	title: 'Atoms/Checkbox',
	component: N8nCheckbox,
};

const methods = {
	onInput: action('input'),
	onFocus: action('focus'),
	onChange: action('change'),
};

const DefaultTemplate: StoryFn = (args, { argTypes }) => ({
	props: Object.keys(argTypes),
	components: {
		N8nCheckbox,
	},
	data: () => ({
		isChecked: false,
	}),
	template: '<n8n-checkbox v-model="isChecked" v-bind="$props" @input="onInput"></n8n-checkbox>',
	methods,
});

export const Default = DefaultTemplate.bind({});
Default.args = {
	label: 'This is a default checkbox',
	tooltipText: 'Checkbox tooltip',
	disabled: false,
	indeterminate: false,
	size: 'small',
};
