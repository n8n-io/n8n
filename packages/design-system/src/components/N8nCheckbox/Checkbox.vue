<template>
	<el-checkbox
		v-bind="$props"
		:class="$style.n8nCheckbox"
		:disabled="disabled"
		:indeterminate="indeterminate"
		:value="value"
		@change="onChange"
	>
		<n8n-input-label
			:label="label"
			:tooltipText="tooltipText"
			:bold="false"
			:size="labelSize"
		></n8n-input-label>
	</el-checkbox>
</template>

<script lang="ts">
import Vue from 'vue';
import ElCheckbox from 'element-ui/lib/checkbox';
import N8nInputLabel from '../N8nInputLabel';

export default Vue.extend({
	name: 'n8n-checkbox',
	components: {
		ElCheckbox,
		N8nInputLabel,
	},
	props: {
		label: {
			type: String,
			required: true,
		},
		disabled: {
			type: Boolean,
			default: false,
		},
		tooltipText: {
			type: String,
			required: false,
		},
		indeterminate: {
			type: Boolean,
			default: false,
		},
		value: {
			type: Boolean,
			default: false,
		},
		labelSize: {
			type: String,
			default: 'medium',
			validator: (value: string): boolean =>
				['small', 'medium'].includes(value),
		},
	},
	methods: {
		onChange(e) {
			this.$emit("input", e);
		},
	}
});
</script>

<style lang="scss" module>

.n8nCheckbox {
	display: flex !important;
	white-space: normal !important;

	span {
		white-space: normal;
	}
}

</style>
