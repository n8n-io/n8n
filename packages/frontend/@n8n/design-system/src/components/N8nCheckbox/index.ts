import N8nCheckbox from './Checkbox.vue';

export default N8nCheckbox;
