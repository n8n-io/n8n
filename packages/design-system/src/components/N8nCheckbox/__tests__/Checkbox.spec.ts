import { render } from '@testing-library/vue';
import N8nCheckbox from '../Checkbox.vue';

describe('components', () => {
	describe('N8nCheckbox', () => {
		it('should render without label and child content', () => {
			const { container } = render(N8nCheckbox);
			expect(container).toMatchSnapshot();
		});

		it('should render with label', () => {
			const { container } = render(N8nCheckbox, { props: { label: 'Checkbox' } });
			expect(container).toMatchSnapshot();
		});

		it('should render with child', () => {
			const { container } = render(N8nCheckbox, {
				slots: { default: '<strong>Bold text</strong>' },
			});
			expect(container).toMatchSnapshot();
		});

		it('should render with both child and label', () => {
			const { container } = render(N8nCheckbox, {
				props: { label: 'Checkbox' },
				slots: { default: '<strong>Bold text</strong>' },
			});
			expect(container).toMatchSnapshot();
		});
	});
});
