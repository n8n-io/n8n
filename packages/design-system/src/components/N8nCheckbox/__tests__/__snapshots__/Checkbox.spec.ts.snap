// Vitest Snapshot v1

exports[`components > N8nCheckbox > should render with both child and label 1`] = `
<div>
  <label
    class="el-checkbox n8n-checkbox n8nCheckbox"
    labelsize="medium"
  >
    <span
      class="el-checkbox__input"
    >
      <span
        class="el-checkbox__inner"
      />
      <input
        aria-hidden="false"
        class="el-checkbox__original"
        type="checkbox"
        value="Checkbox"
      />
    </span>
    <span
      class="el-checkbox__label"
    >
      <strong>
        Bold text
      </strong>
      <div
        class="container"
        data-test-id="input-label"
      >
        <label
          class="n8n-input-label inputLabel heading medium"
        >
          <div
            class="title"
          >
            <span
              class="n8n-text size-medium regular"
            >
               Checkbox 
              <!---->
            </span>
          </div>
          <!---->
          <!---->
          <!---->
        </label>
      </div>
      <!---->
    </span>
  </label>
</div>
`;

exports[`components > N8nCheckbox > should render with child 1`] = `
<div>
  <label
    class="el-checkbox n8n-checkbox n8nCheckbox"
    labelsize="medium"
  >
    <span
      class="el-checkbox__input"
    >
      <span
        class="el-checkbox__inner"
      />
      <input
        aria-hidden="false"
        class="el-checkbox__original"
        type="checkbox"
        value=""
      />
    </span>
    <span
      class="el-checkbox__label"
    >
      <strong>
        Bold text
      </strong>
      <!---->
      <!---->
    </span>
  </label>
</div>
`;

exports[`components > N8nCheckbox > should render with label 1`] = `
<div>
  <label
    class="el-checkbox n8n-checkbox n8nCheckbox"
    labelsize="medium"
  >
    <span
      class="el-checkbox__input"
    >
      <span
        class="el-checkbox__inner"
      />
      <input
        aria-hidden="false"
        class="el-checkbox__original"
        type="checkbox"
        value="Checkbox"
      />
    </span>
    <span
      class="el-checkbox__label"
    >
      <div
        class="container"
        data-test-id="input-label"
      >
        <label
          class="n8n-input-label inputLabel heading medium"
        >
          <div
            class="title"
          >
            <span
              class="n8n-text size-medium regular"
            >
               Checkbox 
              <!---->
            </span>
          </div>
          <!---->
          <!---->
          <!---->
        </label>
      </div>
      <!---->
    </span>
  </label>
</div>
`;

exports[`components > N8nCheckbox > should render without label and child content 1`] = `
<div>
  <label
    class="el-checkbox n8n-checkbox n8nCheckbox"
    labelsize="medium"
  >
    <span
      class="el-checkbox__input"
    >
      <span
        class="el-checkbox__inner"
      />
      <input
        aria-hidden="false"
        class="el-checkbox__original"
        type="checkbox"
        value=""
      />
    </span>
    <!---->
  </label>
</div>
`;
